test/e2e/logging_test.go

```diff
@@ -0,0 +1,183 @@
+// +build e2e
+
+/*
+Copyright 2020 The Knative Authors
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package e2e
+
+import (
+	"bytes"
+	"encoding/json"
+	"fmt"
+	"io"
+	"strings"
+	"testing"
+	"time"
+
+	corev1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/util/wait"
+	"knative.dev/pkg/system"
+	pkgTest "knative.dev/pkg/test"
+	"knative.dev/pkg/test/logstream"
+	"knative.dev/serving/pkg/apis/autoscaling"
+	"knative.dev/serving/pkg/network"
+	rtesting "knative.dev/serving/pkg/testing/v1"
+	"knative.dev/serving/test"
+	v1test "knative.dev/serving/test/v1"
+)
+
+const request_log_template = `
+{"requestMethod": "{{.Request.Method}}", "requestUrl": "{{js .Request.RequestURI}}", "userAgent": "{{js .Request.UserAgent}}"}
+`
+
+func TestRequestLogs(t *testing.T) {
+	t.Parallel()
+	cancel := logstream.Start(t)
+	defer cancel()
+
+	clients := Setup(t)
+
+	// Patch config-observability to enable request logs
+	obcm, err := rawCM(clients, "config-observability")
+	if err != nil {
+		t.Errorf("Error retrieving observability configmap: %v", err)
+	}
```

Comment:
I think we're trying not to do this anymore. 
This one is more obscure and probably does not deserve a separate test suite...
/cc @mattmoor

Reply:
What do you mean by this? Fetching CM or patching CM?

Reply:
patching.

Reply:
Then what to do? Just enable it in the configmap for all tests?

Also do you suggest add the logs verification into an existing e2e test, say `helloworld_test.go`?

Reply:
we usually split this into a separate test suite, but this seems like an overkill.
I think it's probably fine, given it's pretty obscure flag.
I'll let Matt do the final judgement.